src/DGmethods/DGmodel.jl

```diff
@@ -140,25 +166,30 @@ function (dg::DGModel)(dQdt, Q, ::Nothing, t; increment = false)
             grid.vmap⁺,
             grid.elemtobndy,
             hypervisc_indexmap,
-            topology.realelems;
-            ndrange = ndrange_surface,
-            dependencies = (event,),
+            grid.exteriorelems;
+            ndrange = ndrange_exterior_surface,
+            dependencies = (comp_stream, exchange_Q),
         )
-        wait(device, event)
 
         if communicate
-            nviscstate > 0 && MPIStateArrays.start_ghost_exchange!(Qvisc)
-            nhyperviscstate > 0 &&
-            MPIStateArrays.start_ghost_exchange!(Qhypervisc_grad)
+            if nviscstate > 0
+                exchange_Qvisc = MPIStateArrays.begin_ghost_exchange!(
+                    Qvisc,
+                    dependencies = comp_stream,
+                )
+            end
+            if nhyperviscstate > 0
+                exchange_Qhypervisc_grad = MPIStateArrays.begin_ghost_exchange!(
+                    Qhypervisc_grad,
+                    dependencies = comp_stream,
+                )
+            end
         end
 
         if nviscstate > 0
-            aux_comm = update_aux_diffusive!(dg, bl, Q, t)
-            @assert typeof(aux_comm) == Bool
-        end
-
-        if aux_comm
-            MPIStateArrays.start_ghost_exchange!(auxstate)
+            wait(device, comp_stream)
```

Comment:
```suggestion
            # update_aux_diffusive may start asynchronous work on the compute device
            # and we synchronize those here through a device event.
            wait(device, comp_stream)
```

Reply:
Added some comments to hopefully make this a little more clear.